### PR TITLE
Release Automation

### PR DESCRIPTION
**Pull request by jared-s**
# Release Automation

First pass at automation of current crate release workflow. This script uses a 'release plan' that simply consists of a file with crate names and bump level (minor or patch).

For example:
```
ockam_core minor
ockam_node patch
ockam_vault minor
ockam minor
```

When this plan is given to `release.sh`, the following occurs:

- Changes since the last released are gathered from git into a `Changelog-INCOMING.md` for each updated crate. This file has the correct version and date populated, as well as section headers. All that a maintainer needs to do is place the git log message content into one of the categories, then cut and paste into the real `Changelog.md`

- The version of the crate is incremented using `cargo bump`.
- The `README.md` of the crate is updated to the correct version.
- For all other Ockam crates, dependencies on the releasing crate are updated.
- Lockfiles are regenerated for all changed crates.
- All crates are checked via `cargo check`  and  will fail on problems
- Runs a final `gradle test_rust`.

### CODE REVIEW

File: tools/scripts/release/common.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+function all_crates {
+    change_dir "$OCKAM_RUST"
+    for CRATE in *
+    do
+      change_dir "$CRATE"
+      cargo -q "$1" 1>/dev/null || exit 1
+      pop_dir
+    done
+    pop_dir
+}
```
Comment (mrinalwadhwa) [inline tools/scripts/release/common.sh]: please add a trailing empty line

File: tools/scripts/release/release.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    echo "Generating lock files"
+    all_crates generate-lockfile
+  done < "${1:-/dev/stdin}"
+
+  echo "Checking all crates"
+  all_crates check
+pop_dir
+
+echo "All updates complete. Testing"
+"$PWD/gradlew" test_rust
```
Comment (mrinalwadhwa) [inline tools/scripts/release/release.sh]: please add a trailing empty line

**Review by mrinalwadhwa — CHANGES_REQUESTED** [review]
the scripts look great 👍 
left a couple of minor requests for trailing new lines.

**Comment by jared-s** [comment]
Thanks! Corrected all.

**Pull request merged**